packages/opencrvs/test/Adaptor.test.js

```diff
@@ -0,0 +1,196 @@
+
+const authUrl = enableMockClient('https://auth.fake.opencrvs.server.com');
+const baseUrl = 'https://gateway.fake.opencrvs.server.com';
+const testServer = enableMockClient(baseUrl);
+
+const state = {
+  configuration: {
+    domain: 'gateway.fake.opencrvs.server.com',
+    clientId: 'someclientid',
+    clientSecret: 'someclientsecret',
+  },
+};
+
+before(() => {
+  state.configuration.domain = 'fake.opencrvs.server.com';
+  authUrl
+    .intercept({
+      path: '/token',
+      method: 'POST',
+      query: {
+        client_id: state.configuration.clientId,
+        client_secret: state.configuration.clientSecret,
+        grant_type: 'client_credentials',
+      },
+    })
+    .reply(200, {
+      access_token: 'fake-token',
+    })
+    .persist();
+});
+
+describe('request', () => {
+  it('makes a post request to create a birth notification', async () => {
+    testServer
+      .intercept({
+        path: '/notification',
+        method: 'POST',
+        headers: {
+          Authorization: 'Bearer fake-token',
+          'Content-type': 'application/json',
+        },
+      })
+
+      .reply(200, {
+        resourceType: 'Bundle',
+        type: 'document',
+        entry: birthRecordData,
+      });
+    state.data = birthRecordData;
+
+    const finalState = await execute(
+      request('POST', '/notification', {
+        resourceType: 'Bundle',
+        type: 'document',
+        meta: {
+          lastUpdated: new Date().toISOString(),
+        },
+        entry: birthRecordData,
+      })
+    )(state);
+
+    expect(finalState.data).to.eql({
+      resourceType: 'Bundle',
+      type: 'document',
+      entry: birthRecordData,
+    });
+  });
+});
+
+describe('post', () => {
+  it('makes a post request to create a birth notification', async () => {
+    testServer
+      .intercept({
+        path: '/notification',
+        method: 'POST',
+        headers: {
+          Authorization: 'Bearer fake-token',
+          'Content-type': 'application/json',
+        },
+      })
+
+      .reply(200, {
+        resourceType: 'Bundle',
+        type: 'document',
+        entry: birthRecordData,
+      });
+    state.data = birthRecordData;
+
+    const finalState = await execute(
+      post('/notification', {
+        resourceType: 'Bundle',
+        type: 'document',
+        meta: {
+          lastUpdated: new Date().toISOString(),
+        },
+        entry: birthRecordData,
+      })
+    )(state);
+
+    expect(finalState.data).to.eql({
+      resourceType: 'Bundle',
+      type: 'document',
+      entry: birthRecordData,
+    });
+  });
+});
+
+describe('createBirthRecord', () => {
+  it('successfully creates a birth record', async () => {
+    testServer
+      .intercept({
+        path: '/notification',
+        method: 'POST',
+        headers: {
+          Authorization: 'Bearer fake-token',
+          'Content-type': 'application/json',
+        },
+      })
+
+      .reply(200, {
```

Comment:
I would like the reply handler in this test to assert that a bundle has been uploaded in the correct format.

Because if you change the `createBirthNotification` to not use a bundle, then the real OpenCRVS endpoint will fail, but the test will still pass. It's not strict enough.

So I guess I'm saying that in THIS test, you need to simulate the logic of the server a little bit. Because you are now testing that this endpoint _specifically_ is being called correctly.